### PR TITLE
Move hidePromotedTweets to new file

### PR DESCRIPTION
**Pull request by paulmolluzzo**
Using @SamVerschueren's suggestion [here](https://github.com/sindresorhus/refined-twitter/pull/41#discussion_r67021794) I broke this into a new file that is identical to the file used in `refined-twitter` here: https://github.com/sindresorhus/refined-twitter/pull/41

### CODE REVIEW

File: browser.js [hunk 1]
```diff
@@ -23,7 +24,7 @@ function changeTab(next) {
 }
 
 // sets interval to wait for selector to be ready before firing callback
-function waitFor(selector) {
+const waitFor = selector => {
```
Comment (paulmolluzzo) [inline browser.js:27]: @sindresorhus is this actually required? I made the change and didn't even think if I need to or not.

Comment (sindresorhus) [inline browser.js:27]: Not needed, but it's fine.

File: vendor/hide-promoted-tweets.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+			new MutationObserver(() => {
+				seekAndDestroy();
+			}).observe(tweetContainer, {attributes: true, childList: true});
+		});
+	};
+
+	if (typeof module !== 'undefined' && module.exports) {
+		module.exports = window.hidePromotedTweets;
+	}
+})(window || module.exports);
```
Comment (sindresorhus) [inline vendor/hide-promoted-tweets.js]:
``` js
})();
```

You're already assigning it to module.exports inline, which is better.

File: vendor/hide-promoted-tweets.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
+(exports => {
+	'use strict';
+
+	exports.hidePromotedTweets = waitFor => {
```
Comment (sindresorhus) [inline vendor/hide-promoted-tweets.js]:
``` js
window.hidePromotedTweets
```

File: vendor/hide-promoted-tweets.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
+(exports => {
```
Comment (sindresorhus) [inline vendor/hide-promoted-tweets.js]:
``` js
(() => {
```

**Comment by paulmolluzzo** [comment]
@sindresorhus fixed and pushed.

File: vendor/hide-promoted-tweets.js (hunk 2 above)
Comment (SamVerschueren) [inline vendor/hide-promoted-tweets.js]: Just wondering why it's better? My solution looks a little cleaner to me but might not cover all use cases?

Comment (SamVerschueren) [inline vendor/hide-promoted-tweets.js]: Alright, noticed your comment in the other repo. :+1:

**Comment by sindresorhus** [comment]
👌

**Pull request merged**